interpreter/utils/get_config.py

```diff
@@ -1,25 +1,50 @@
 import os
 import yaml
-import appdirs
 from importlib import resources
 import shutil
 
+from .local_storage_path import get_storage_path
+
 config_filename = "config.yaml"
 
-# Using appdirs to determine user-specific config path
-config_dir = appdirs.user_config_dir("Open Interpreter")
-user_config_path = os.path.join(config_dir, config_filename)
-
-def get_config():
-    if not os.path.exists(user_config_path):
-        # If user's config doesn't exist, copy the default config from the package
-        here = os.path.abspath(os.path.dirname(__file__))
-        parent_dir = os.path.dirname(here)
-        default_config_path = os.path.join(parent_dir, 'config.yaml')
-        # Ensure the user-specific directory exists
-        os.makedirs(config_dir, exist_ok=True)
-        # Copying the file using shutil.copy
-        shutil.copy(default_config_path, user_config_path)
-
-    with open(user_config_path, 'r') as file:
+user_config_path = os.path.join(get_storage_path(), config_filename)
+
+def get_config_path(path=user_config_path):
+    # check to see if we were given a path that exists
+    if not os.path.exists(path):
+        # check to see if we were given a filename that exists in the config directory
+        if os.path.exists(os.path.join(get_storage_path(), path)):
+            path = os.path.join(get_storage_path(), path)
+        else:
+            # check to see if we were given a filename that exists in the current directory
+            if os.path.exists(os.path.join(os.getcwd(), path)):
+                path = os.path.join(os.path.curdir, path)
+            # if we weren't given a path that exists, we'll create a new file
+            else:
+                # if the user gave us a path that isn't our default config directory
+                # but doesn't already exist, let's create it
+                if os.path.dirname(path) and not os.path.exists(os.path.dirname(path)):
+                    os.makedirs(os.path.dirname(path), exist_ok=True)
+                else:
+                    # Ensure the user-specific directory exists
+                    os.makedirs(get_storage_path(), exist_ok=True)
+                    
+                    # otherwise, we'll create the file in our default config directory
+                    path = os.path.join(get_storage_path(), path)
+
+
+                # If user's config doesn't exist, copy the default config from the package
+                here = os.path.abspath(os.path.dirname(__file__))
+                parent_dir = os.path.dirname(here)
+                default_config_path = os.path.join(parent_dir, 'config.yaml')
+
+                # Copying the file using shutil.copy
+                new_file = shutil.copy(default_config_path, path)
```

Comment:
I was thinking that we might want to exclude the `system_message` from the new `config.yaml` file in order to have people using the default message as things are upgraded unless they _really_ want to customize it.

Right now any changes we make to the default `system_message` in [`interpeter/config.yaml`](https://github.com/KillianLucas/open-interpreter/blob/main/interpreter/config.yaml#L1) will be overwritten by the `system_message` in the user's `config.yaml` file, which means if we need to push out some update to the underlying prompt engineering that powers Open Interpreter, users who have already generated a `config.yaml` file will not receive it unless they manually update their `config.yaml` with the latest system prompt.

It might also be good to introduce a property like `append_to_sysem_message` or `appended_system` message that will allow the user to just specify some instructions they want to concatenate onto the end of our default `system_message`.

Reply:
Makes sense actually, didn't think of when we would update the system_prompt from our ends.